kernel/kernel-api/src/main/java/io/delta/kernel/expressions/CastingComparator.java

```diff
@@ -0,0 +1,48 @@
+package io.delta.kernel.expressions;
+
+import java.util.Comparator;
+
+import io.delta.kernel.types.*;
+import io.delta.kernel.types.BooleanType;
+import io.delta.kernel.types.DataType;
+import io.delta.kernel.types.IntegerType;
+import io.delta.kernel.types.LongType;
+import io.delta.kernel.types.StringType;
+
+// TODO: exclude from public interfaces (move to "internal" somewhere?)
```

Comment:
remove?

Reply:
I think we need to do this in the future? CastingComparator shouldn't be a public API right? It's part of our own impl. of eval